core/src/main/java/com/wttech/aem/acm/core/servlet/StateServlet.java

```diff
@@ -42,12 +49,21 @@ public class StateServlet extends SlingAllMethodsServlet {
     @Reference
     private ExecutionQueue executionQueue;
 
+    @Reference
+    private OsgiContext osgiContext;
+
+    @Reference
+    private SlingSettingsService slingSettingsService;
+
     @Override
     protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
         try {
             HealthStatus healthStatus = healthChecker.checkStatus();
-            InstanceSettings instanceSettings = InstanceSettings.current();
             List<Execution> queuedExecutions = executionQueue.findAll().collect(Collectors.toList());
+            boolean publish = osgiContext.getInstanceInfo().isPublish();
+            String propUrl = osgiContext.getBundleContext().getProperty("PRODUCTINFO_VERSION");
```

Comment:
please don't do too much in servlet; treat it like controller in MVC framework; it should only handle input/output and mostly delegate things to services;

this code is too low level for servlet / IMO